File: 20240812/Q1.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	int i, j;
    string str[2];

    cout << "Please input string #1: ";
    getline(cin, str[0]);
    cout << "Please input string #2: ";
    getline(cin, str[1]);

    int counts[2][26] = {{0}};

    for (i = 0; i < 2; i++) {
        for (j = 0; j < str[i].length(); j++) {
            char c = str[i][j];

            if ('A' <= c && c <= 'Z') {
                counts[i][c - 'A'] ++;
            } else if ('a' <= c && c <= 'z') {
                counts[i][c - 'a'] ++;
            }
        }
    }

    for (i = 0; i < 26; i++) {
        if (counts[0][i] != counts[1][i]) {
            break;
        }
    }

    if (i == 26) {
        cout << "Yes" << endl;
    } else {
        cout << "No" << endl;
    }

    return 0;
}

File: 20240812/Q1.py
str1 = input("Please input string #1:")
str2 = input("Please input string #2:")

counts = [[0] * 26, [0] * 26]

for i, s in enumerate([str1, str2]):
    for char in s:
        if 'A' <= char <= 'Z':
            counts[i][ord(char) - ord('A')] += 1
        elif 'a' <= char <= 'z':
            counts[i][ord(char) - ord('a')] += 1

if counts[0] == counts[1]:
    print("Yes")
else:
    print("No")
